src/risedevtool/src/bin/risedev-docslt.rs

```diff
@@ -0,0 +1,251 @@
+            "--document-private-items",
+        ])
+        .output()?;
+    if !rustdoc_output.status.success() {
+        let stderr = String::from_utf8(rustdoc_output.stderr)?;
+        if stderr.contains("no library targets found") {
+            println!("{}", style("IGNORED").yellow().bold());
+            return Ok(());
+        } else {
+            println!("{}", style("FAILED").red().bold());
+            println!("{}\n{}", style("Error output:").red().bold(), stderr);
+            return Err(anyhow!(
+                "`cargo rustdoc` failed with exit code {}",
+                rustdoc_output.status.code().unwrap()
+            ));
+        }
+    }
+
+    let rustdoc: JsonValue = serde_json::from_reader(std::io::BufReader::new(
+        std::fs::File::open(format!("target/doc/{}.json", package_name))?,
+    ))?;
+    let index = rustdoc["index"]
+        .as_object()
+        .ok_or_else(|| anyhow!("failed to access `index` field as object"))?;
+
+    // slt blocks in each file
+    let mut slt_blocks_per_file: HashMap<String, Vec<SltBlock>> = HashMap::new();
+
+    for item in index.values() {
+        let docs = item["docs"].as_str().unwrap_or("");
+        if docs.contains("```slt") {
+            let filename = item["span"]["filename"]
+                .as_str()
+                .ok_or_else(|| anyhow!("docslt blocks are expected to be in files"))?;
+            let line_no = item["span"]["begin"][0]
+                .as_u64()
+                .ok_or_else(|| anyhow!("docslt blocks are expected to have line number"))?
+                as usize;
+            let item_name = item["name"].as_str().unwrap_or("").to_string();
+            let slt_blocks = extract_slt_blocks(
+                docs,
+                FilePosition {
+                    filepath: PathBuf::from(filename),
+                    line_no,
+                    item_name,
+                },
+            );
+            if slt_blocks.is_empty() {
+                continue;
+            }
+            slt_blocks_per_file
+                .entry(filename.to_string())
+                .or_default()
+                .extend(slt_blocks);
+        }
+    }
+
+    for slt_blocks in slt_blocks_per_file.values_mut() {
+        slt_blocks.sort_by_key(|block| (block.typ, block.position.line_no));
+    }
+
+    let slt_dir = PathBuf::from(format!("e2e_test/generated/docslt/{}", package_name));
+    std::fs::remove_dir_all(&slt_dir).ok();
+    if !slt_blocks_per_file.is_empty() {
+        std::fs::create_dir_all(&slt_dir)?;
+    }
+
+    for filename in slt_blocks_per_file.keys() {
+        let filename_digest = format!("{:x}", md5::compute(filename));
+        let file_name = Path::new(filename).file_name().unwrap().to_str().unwrap();
+        let file_basename = Path::new(file_name)
+            .file_prefix()
+            .unwrap()
+            .to_str()
+            .unwrap();
+        let slt_filename = format!("{file_basename}_{filename_digest}.slt");
```

Comment:
Why to use digest in filename? 🤔 What about simply use dirs or replace `/` with `_`?

Reply:
That will make the filename too long? There're too many `src` in the path🤣